common/src/main/java/tachyon/network/protocol/RPCBlockWriteRequest.java

```diff
@@ -59,13 +59,10 @@ public static RPCBlockWriteRequest decode(ByteBuf in) {
     long blockId = in.readLong();
     long offset = in.readLong();
     long length = in.readLong();
-    DataBuffer data = null;
-    if (length > 0) {
-      // TODO(hy): Look into accessing Netty ByteBuf directly, to avoid copying the data.
-      ByteBuffer buffer = ByteBuffer.allocate((int) length);
-      in.readBytes(buffer);
-      data = new DataByteBuffer(buffer, (int) length);
-    }
+    // TODO(hy): Look into accessing Netty ByteBuf directly, to avoid copying the data.
```

Comment:
could you comment here that `length` can be zero and we still allocate the buffer and make the request, and also state the assumption we have here for `DataServerHandler`?